test/parallel/test-module-main-extensions.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const { execFileSync } = require('child_process');
+
+const node = process.argv[0];
+
+try {
+  execFileSync(node, ['--experimental-modules', 'test/es-module/test-esm-ok']);
+} catch (e) {
```

Comment:
Should this ever throw an exception?

Reply:
The resolver can throw but it shouldn't... let me clean this up.